src/main/java/phi/TaskList.java

```diff
@@ -0,0 +1,174 @@
+package phi;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Represents the task list of the chatbot
+ */
+public class TaskList {
+    private final ArrayList<Task> tasks;
+
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    private Task getTask(int i) {
+        if (i <= 0 || i > tasks.size()) {
+            throw new IllegalArgumentException("this task does not exist, genius..");
+        } else {
+            return tasks.get(i - 1);
+        }
+    }
+
+    /**
+     * Marks a task as done and returns the corresponding response message
+     * Contains error handling for incorrect user inputs
+     *
+     * @param input User-inputted string to be parsed
+     * @return      String response
+     */
+    public String doTask(String input) {
```

Comment:
markDone() method above and doTask(String) are performing similar actions and can be confusing. Consider renaming or re-organizing the computation elsewhere, keeping only one do function in Task class.